ngx-tools/src/update-control-on-input-changes/update-control-on-input-changes.ts

```diff
@@ -20,9 +20,12 @@ export function updateControlOnInputChanges(
   if (!changes || !key || !control) {
     return false;
   }
+  let fControl = '';
+  let field = '';
 
-  if (inputHasChanged(changes, key)) {
-    control.setValue(changes[key].currentValue);
+  [fControl, field] = key.split('.');
```

Comment:
You can combine lines 23-26: `const [fControl, field]: string[] = key.split('.');`

Reply:
We should probably support more than a single path deep. I can see us hitting that pretty quickly once this functionality is partially supported. (plus it can be confusing for consumers since the `foo.bar` format is supported but only one level deep)

`updateControlOnInputChanges(changes, 'foo.bar.baz', control)`

Possibly change the `key` param to `path` so it's clear we are using that to find specific data in the object.

Reply:
Looking through this more, we may want to move all the key/path logic to `inputHasChanged` since I believe that function will also need to deep-dive into the object for a specific value. That way we can hopefully only do it in one place. 

Once this change is made, the params `key` and `field` from `inputHasChanged` can be combined into a single param:

```typescript
updateControlOnInputChanges(changes, 'foo.bar.baz', control) {
  const hasChanged = inputHasChanged(changes, path);
}
```

This will still be non-breaking for existing usage.

Reply:
Even if we move all the parse logic to `inputHasChanged`, `updateControlOnInputChanges` still needs that separation info to `setValue`, right?